Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -86,9 +92,22 @@ $msDeployCmdArgs = Get-MsDeployCmdArgs -packageFile $packageFile -webAppNameForM
 # Deploy azureRM webApp using msdeploy Command
 Run-MsDeployCommand -msDeployExePath $msDeployExePath -msDeployCmdArgs $msDeployCmdArgs
 
-# Publish azure webApp url
-$azureWebsitePublishURL = Get-WebAppPublishURL -webAppName $WebAppName -deployToSlotFlag $DeployToSlotFlag -slotName $SlotName
+# Get azure webapp hosted url
+$azureWebsitePublishURL = Get-AzureRMWebAppPublishUrl -webAppName $WebAppName -deployToSlotFlag $DeployToSlotFlag `
+                                                                       -resourceGroupName $ResourceGroupName -slotName $SlotName
+
+if( [string]::IsNullOrEmpty($azureWebsitePublishURL))
+{
+    Throw (Get-LocalizedString -Key "Unable to retrieve webapp publish url for webapp : '{0}'." -ArgumentList $webAppName)
```

Comment:
Should this check be done only if $WebAppUri output param is set? I don't see a reason to throw otherwise.